### PR TITLE
No need to roundtrip symbol when doing OOP rename call.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Found while doing a larger refactoring.  Extracted out for simplicity.

### CODE REVIEW

File: src/Workspaces/Core/Portable/Rename/RenameLocations.cs [hunk 1]
```diff
@@ -92,7 +92,7 @@ internal sealed partial class RenameLocations
                         if (result.HasValue && result.Value != null)
                         {
                             var rehydrated = await TryRehydrateAsync(
-                                solution, fallbackOptions, result.Value, cancellationToken).ConfigureAwait(false);
+                                solution, symbol, fallbackOptions, result.Value, cancellationToken).ConfigureAwait(false);
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Rename/RenameLocations.cs:95]: this is the core change. we are already passing the symbol in, in order to make the oop call.  so no need to try to have dehydrate/rehydrate that exact same symbol in its response when we can just literally use the same one we started with.

File: src/Workspaces/Remote/ServiceHub/Services/Renamer/RemoteRenamerService.cs [hunk 2]
```diff
@@ -85,26 +85,31 @@ private CodeCleanupOptionsProvider GetClientOptionsProvider(RemoteServiceCallbac
 
                 var result = await RenameLocations.FindLocationsAsync(
                     symbol, solution, options, fallbackOptions, cancellationToken).ConfigureAwait(false);
-
                 return result.Dehydrate(solution, cancellationToken);
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Remote/ServiceHub/Services/Renamer/RemoteRenamerService.cs:88]:
```suggestion

                return result.Dehydrate(solution, cancellationToken);
```

**Review by Cosifne — APPROVED** [review]

File: src/Workspaces/Core/Portable/Rename/IRemoteRenamerService.cs [hunk 3]
```diff
@@ -74,64 +75,6 @@ public ValueTask<CodeCleanupOptions> GetOptionsAsync(RemoteServiceCallbackId cal
             => ((RemoteOptionsProvider<CodeCleanupOptions>)GetCallback(callbackId)).GetOptionsAsync(language, cancellationToken);
     }
 
-    [DataContract]
-    internal class SerializableSearchResult
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Rename/IRemoteRenamerService.cs:78]: removed thsi concept entirely.  a SerializableRenameLocations points at a single SerializableSearchResult.  i've just inlined this for simplicity.

**Comment by Cosifne** [comment]
It makes sense to save one symbol's work. But I wonder how about the other symbols in the symbolset?
Is the ultimate goal to have a remote 'session' in the OOP to store the rename locations so the communication between OOP and devenv would just be some textspans and doucmentIds?

**Comment by CyrusNajmabadi** [comment]
>Is the ultimate goal to have a remote 'session' in the OOP to store the rename locations so the communication between OOP and devenv would just be some textspans and doucmentIds?

Not quite.  I don't intend OOP to be stateful.  But i do intend taht all data send to/from OOP is lightweight and will not cause symbol resolution to have to happen again on the VS side.

**Pull request merged**